starlark/src/syntax/grammar_tests.rs

```diff
@@ -226,9 +226,8 @@ fn smoke_test() {
     let paths = fs::read_dir(d.as_path()).unwrap();
     for p in paths {
         let path_entry = p.unwrap().path();
-        let path = path_entry.to_str().unwrap();
-        if path.ends_with(".bzl") {
-            if let Result::Err(err) = parse_file(&map, path, Dialect::Bzl) {
+        if path_entry.ends_with(".bzl") {
```

Comment:
`ends_with` on a `Path` actually compares components, rather than bytes, so this is asking if the last path component is `.bzl` (i.e. the path is `some/prefix/.bzl`), which isn't quite what we want to do here.

I think:

```rust
if let Some(extension) = path_entry.extension() {
    if extension == ".bzl" {
```

would do what you're trying to do :)

Reply:
Yep! Turns out i missed this when find/replace/fix compile errors. Whoops!